application/core.py

```diff
@@ -0,0 +1,236 @@
+        for F in (MainMenu,):
+            page_name = F.__name__
+            frame = F(parent=container, controller=self)
+            self.frames[page_name] = frame
+            frame.grid(row=0, column=0, sticky="nsew")
+
+        self.show_frame("MainMenu")
+        self.title("async2rewrite")
+
+    def show_frame(self, page_name):
+        """Show a frame for the given page name."""
+        frame = self.frames[page_name]
+        frame.tkraise()
+
+
+class MainMenu(tk.Frame):
+    """Main menu of the application."""
+    def __init__(self, parent, controller):
+        super().__init__(parent)
+
+        # Basic declarations
+        self.parent = parent
+        self.controller = controller
+
+        # Label
+        a2r_img = tk.PhotoImage(file='../logo.png')
+        label = tk.Label(self, image=a2r_img)
+        label.image = a2r_img
+        label.grid(row=1, column=1, sticky="we")
+
+        # Menu buttons
+
+        snippet = tk.Button(self, text="Convert Snippet", command=lambda: Snippet(self.parent))
+        snippet.grid(row=2, column=1, sticky="we")
+
+        file = tk.Button(self, text="Convert File", command=self.file)
+        file.grid(row=3, column=1, sticky="we")
+
+        notices = tk.Button(self, text="Credits", command=lambda: Notices(self.parent))
+        notices.grid(row=4, column=1, sticky="we")
+
+        # Configure grid geometry
+        self.grid_rowconfigure(1, weight=1)
+        self.grid_columnconfigure(1, weight=1)
+
+    def file(self):
+        """Converts a file from async to rewrite."""
+        name = filedialog.askopenfilename(initialdir=".", title="Select a file",
+                                          filetypes=(("Python Files", "*.py"),))  # Asks for original file
+        save = filedialog.asksaveasfilename(initialdir=".", title="Save file as",
+                                            filetypes=(("Python Files", "*.py"),))  # Asks where to save the file
+
+        try:
+            file_result = async2rewrite.from_file(name)  # Converts code from original file to rewrite
+            with open(save, 'w') as f:
+                f.write(file_result.replace('\t', '    '))
+        except Exception as e:
+
+            # Check if it's to do with file/directory not being found
+            if 'No such file or directory' in str(e):
+                messagebox.showerror('File Not Found', 'You did not choose a file to convert/save the conversion.')
+            else:
+                messagebox.showerror(type(e).__name__, str(e).replace('<unknown>', Path(name).name))
+        else:
+            save = Path(save).parent
+
+            # Copy save url to clipboard
+            self.controller.clipboard_clear()
+            self.controller.clipboard_append(save)
+            self.controller.update()
+
+            messagebox.showinfo('Success!', 'The conversion was successful. Your converted file is in "{}"'.format(save)
+                                            + '(Copied to clipboard).')

```

Comment:
I don't particularly like the practice of automatically copying to clipboard. In addition, I don't really see why copying to clipboard is even necessary here. Probably should remove auto copy to clipboard.